MoltenVK/CMakeLists.txt

```diff
@@ -0,0 +1,204 @@
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKInstance.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKPipeline.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKPixelFormats.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKQueryPool.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKQueue.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKRenderPass.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKResource.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKShaderModule.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKSurface.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKSwapchain.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKSync.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKVulkanAPIObject.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Layers/MVKExtensions.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Layers/MVKLayers.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/CAMetalLayer+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MTLRenderPassDepthAttachmentDescriptor+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MTLRenderPassDescriptor+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MTLRenderPassStencilAttachmentDescriptor+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MTLRenderPipelineDescriptor+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MTLSamplerDescriptor+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MTLTextureDescriptor+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MVKGPUCapture.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/NSString+MoltenVK.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility/MVKBaseObject.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility/MVKBlockObserver.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility/MVKCodec.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility/MVKEnvironment.cpp
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility/MVKFoundation.cpp
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility/MVKWatermark.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Vulkan/mvk_datatypes.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Vulkan/mvk_api.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Vulkan/vulkan.mm
+        )
+
+##
+## MoltenVK-iOS
+##
+
+add_library(MoltenVK-iOS STATIC
+        ${MOLTEN_VK_SOURCES}
+        )
+
+target_include_directories(MoltenVK-iOS
+        PUBLIC
+        ${CMAKE_CURRENT_LIST_DIR}/../Common
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/API
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Commands
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Layers
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Vulkan
+        ${CMAKE_CURRENT_LIST_DIR}/include
+        )
```

Comment:
why not also use set to declare a list of include directories upfront, so that we can reuse the same definition for all of the different targets?